wireless_modules/middle_module/MQTT_Client_class.py

```diff
@@ -0,0 +1,89 @@
+# Note: Can remove all the Print statements once all testing is successful
+
+# Use micro pip to install packages
+import upip
+upip.install('micropython-umqtt.simple2')
+try:
+    from umqtt.simple import MQTTClient
+except FileNotFoundError:
+    print('Error in importing library')
+
+
+class Client:
+    def __init__(self, client_id, mqtt_broker):
+        """
+        Initialises the MQTT Client
+        :param client_id: The unique client id used to initiate an MQTTClient class
+        :param mqtt_broker: A string holding domain name or IP address of the broker to connect to, to send and receive
+                            data.
+        """
+        self.client = MQTTClient(client_id, mqtt_broker)
+        self.mqtt_broker = mqtt_broker
+
+    def connect_and_subscribe(self, topics_to_subscribe, callback_func):
+        """
+        Connects to the MQTT broker and subscribes to topic in 'topics_to_subscribe'
+        :param topics_to_subscribe: An array of topics to subscribe to.
+                                    Eah element must be a string or byte literal (the latter is preferred)
+        :param callback_func: The function to be called whenever a message from the subscribed topics is received.
+        :return: True if a connection to the Broker is successfully established, otherwise False
+        """
+
+        self.client.set_callback(callback_func)
+        try:
+            # Connect to MQTT broker
+            self.client.connect()
+            print('Connected to %s MQTT broker' % (self.mqtt_broker))
+        except OSError:
+            return False
+
+        # Subscribe to each topic
+        for topic in topics_to_subscribe:
+            self.client.subscribe(topic)
+            print('Subscribed to %s topic' % (topic))
+
+        return True
+
+    def _to_bytes_literal(self, data):
+        """
+        Converts 'data' into a form MQTT can read
```

Comment:
I'm a bit confused, I don't see any effect of having the backticks around the variable name when I hover over the function. Is this something that only works with viewing the code on GitHub maybe?

Reply:
Might be only in visual studio code. What editor do you use?

Reply:
I have checked it on VS code and PyCharm, just confirming in the screenshot, there's no backticks but the variable name is highlighted right?

Reply:
Oh right, in the code the variable is only highlighted I think because my cursor is inside it, this is just for the documentation that appears when you hover over the function name.